Build docs using GitHub Actions

huwshimi opened the pull request:
## Done

- Remove docs build files from the repository.
- Set up GitHub Actions to publish the docs on release.

## Fixes

Fixes: #944.

huwshimi commented:
Once this lands we'll need to change the setting to use the action to build the docs instead of using the current build files.

bartaz commented on .github/workflows/publish-on-release.yml:
```diff
@@ -23,3 +23,28 @@ jobs:
       - run: npm publish --access public
         env:
           NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+  build-docs:
```
I would add `needs: publish-npm` to make sure we don't update the docs if npm publish fails for any reason.

huwshimi replied:
Good thinking, I've updated that.

bartaz commented:
As this affects the release process there are some docs that may need updating. Like: https://github.com/canonical/react-components/blob/main/PUBLISH-NPM-PACKAGE.md

Also the "prepare-release" script mentioned there is built to automate the `build-docs` step, which is not needed. So the script may need updating (or removing completely).

I guess once we land this, the whole "Prepare release" PR would be reduced to changing version number, which in practice we could move into individual PRs that do the change (as in Vanilla), to remove the need of "Prepare release" PR completely.

Anyway, I'd suggest at least slightly updating the https://github.com/canonical/react-components/blob/main/PUBLISH-NPM-PACKAGE.md docs, to mention the changes, and maybe mention somewhere that docs are built automatically by GH action.

I also wonder if there is still a way to manually build the docs in case we need it outside of the release process.

bartaz reviewed (approved):
Looks good. I'd just add the dependency between docs job and publish to npm job as mentioned in inline comment, and possibly adjust the docs.

huwshimi commented:
> As this affects the release process there are some docs that may need updating. Like: https://github.com/canonical/react-components/blob/main/PUBLISH-NPM-PACKAGE.md

These docs didn't actually mention building the docs or automatically publishing to NPM, so I've added details about both of them.

> 
> Also the "prepare-release" script mentioned there is built to automate the `build-docs` step, which is not needed. So the script may need updating (or removing completely).

In this PR I removed the docs building from `prepare-release` but left the script to update the version, but I think just updating the version might be less confusing than running the script. I've updated the docs to reflect this and removed the `prepare-release` script.

> 
> I guess once we land this, the whole "Prepare release" PR would be reduced to changing version number, which in practice we could move into individual PRs that do the change (as in Vanilla), to remove the need of "Prepare release" PR completely.

I think that's a good idea.

> 
> Anyway, I'd suggest at least slightly updating the https://github.com/canonical/react-components/blob/main/PUBLISH-NPM-PACKAGE.md docs, to mention the changes, and maybe mention somewhere that docs are built automatically by GH action.

Done and done.

> 
> I also wonder if there is still a way to manually build the docs in case we need it outside of the release process.

Yes! You can configure an action to allow it to be manually run. I've done this and added a section to the docs.
